### PR TITLE
fix(forms): getRawValue correctly work with nested form groups

### LINKED ISSUE #12963: getRawValue does not return the raw value of nested controls.
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

If I have a FormGroup *a* with a nested FormGroup *b*, a.getRawValue() only takes the *.value* of *b*, meaning that the values of disabled controls within *b* are not returned.

**Expected behavior**

I expect a.getRawValue() to return all values within the form.

**Minimal reproduction of the problem with instructions**

http://plnkr.co/edit/3VMar1qp8X6mHU6HJToA?p=preview
values are missing from the output.

Updated version with both FormGroup and FormArray:
http://plnkr.co/edit/m9OrFDi3c7qmMoWDtS1A?p=preview

**What is the motivation / use case for changing the behavior?**
Current behaviour is unexpected.

* **Angular version:** 2.1.0

The code looks the same in the latest version though. 
 
* **Language:** [TypeScript]

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes https://github.com/angular/angular/issues/12963

### CODE REVIEW

**Comment by DzmitryShylovich** [comment]
Travis failed because of a Saucelabs flake.

**Comment by beakdoctor** [comment]
It's also `getRawValue` in the `FormArray` class (updated plunkr in #12964 shows the same problem with a `FormArray` and a nested `FormGroup` and `FormArray` within it)

File: modules/@angular/forms/src/model.ts [hunk 1]
```diff
@@ -352,6 +352,22 @@ export abstract class AbstractControl {
...
+    }
+  }
+
+  asGroup(): FormGroup {
+    if (this instanceof FormGroup) {
+      return this as FormGroup;
+    } else {
+      throw new Error(`Expected ${this} to be an instance of FormGroup.`);
+    }
+  }
```
Comment (kara) [inline modules/@angular/forms/src/model.ts]: I think these methods need more discussion, so the bug fix should be separate. Can you move into a separate PR?

File: modules/@angular/forms/test/form_array_spec.ts [hunk 2]
```diff
@@ -32,6 +32,23 @@ export function main() {
...
+          new FormGroup({'c2': new FormControl('v2'), 'c3': new FormControl('v3')}),
+          new FormArray([new FormControl('v4'), new FormControl('v5')])
+        ]);
+        a.at(0).get('c3').disable();
+        a.at(1).asArray().at(1).disable();
+
+        expect(a.getRawValue()).toEqual([{'c2': 'v2', 'c3': 'v3'}, ['v4', 'v5']]);
+      });
+
+    });
```
Comment (kara) [inline modules/@angular/forms/test/form_array_spec.ts]: Nit: can you move this section down below `value`? Kind of weird having it first.

**Review by kara — CHANGES_REQUESTED** [review]

File: modules/@angular/forms/test/form_array_spec.ts (hunk 2 above)
Comment (DzmitryShylovich) [inline modules/@angular/forms/test/form_array_spec.ts]: done

File: modules/@angular/forms/src/model.ts (hunk 1 above)
Comment (DzmitryShylovich) [inline modules/@angular/forms/src/model.ts]: reverted

**Comment by DzmitryShylovich** [comment]
@kara ready for review

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**